okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/UrlSelectorImpl.java

```diff
@@ -40,11 +41,14 @@
     private static final Duration RANDOMIZE = Duration.ofMinutes(10);
 
     private final Supplier<List<HttpUrl>> baseUrls;
-    private final AtomicInteger currentUrl;
+    private final AtomicReference<HttpUrl> currentBaseUrl;
     private final Cache<HttpUrl, UrlAvailability> failedUrls;
     private final boolean useFailedUrlCache;
 
     private UrlSelectorImpl(ImmutableList<HttpUrl> baseUrls, boolean reshuffle, Duration failedUrlCooldown) {
+        Preconditions.checkArgument(!baseUrls.isEmpty(), "Must specify at least one URL");
+        Preconditions.checkArgument(!failedUrlCooldown.isNegative(), "Cache expiration must be non-negative");
```

Comment:
I just moved these at the top, makes reasoning about hte rest of the constructor easier